dotcom-rendering/src/components/TableOfContents.importable.tsx

```diff
@@ -138,13 +138,27 @@ const verticalStyle = css`
 export const TableOfContents = ({ tableOfContents, format }: Props) => {
 	const [open, setOpen] = useState(tableOfContents.length < 5);
 
+	useEffect(() => {
+		const tocElement = document.querySelector(
+			'[data-component="table-of-contents"]',
+		) as HTMLElement;
+
+		const observer = new IntersectionObserver(
```

Comment:
I assume we're checking if the TOC is / isn't in view and setting it to be open / closed respectively?

Would the custom hook [useInView](https://github.com/guardian/dotcom-rendering/blob/a2cc3bb72a36cd666597eb6146826678c721acdf/dotcom-rendering/src/lib/useIsInView.ts) work for this use case? It gives you a `isInView` boolean and a ref to attach to the element you're interested in. You might need to flip the boolean logic here but it might be useful.

Some [examples](https://github.com/search?q=repo%3Aguardian%2Fdotcom-rendering+%22useIsInView%28%22&type=code)

Reply:
Thanks for this @arelra! I've had a go at rewriting this to use `useInView` but I can't get it to work. When the user scrolls beyond the ToC, `useEffect` doesn't get triggered for some reason.

WIP commit here: https://github.com/guardian/dotcom-rendering/pull/13766/commits/0cbc725b6b1da0be35f80db66fba9363ea2bc1a2

Can you spot anything off, or would you be up for pairing?

Reply:
To complete this thread `useInView` required the `repeat` property to trigger more than once. However in order to only close when the element becomes sticky we need to know when the top of the element via `boundingClientRect` becomes negative. So the solution here uses a custom `IntersectionObserver` rather than `useInView`